internal/server/statvar_hierarchy_util.go

```diff
@@ -86,34 +92,42 @@ func GetParentSvgMap(rawSvg map[string]*pb.StatVarGroupNode) map[string][]string
 	return parentSvgMap
 }
 
-func updateSearchIndex(tokenString string, index map[string]map[string]RankingInfo, nodeID string) {
-	processedTokenString := strings.ToLower(tokenString)
+// Update search index, given a stat var (group) node ID and string.
+func (index *SearchIndex) update(
+	nodeID string, nodeString string) {
+	processedTokenString := strings.ToLower(nodeString)
 	processedTokenString = strings.ReplaceAll(processedTokenString, ",", " ")
 	tokenList := strings.Fields(processedTokenString)
 	approxNumPv := len(strings.Split(nodeID, "_"))
 	if approxNumPv == 1 {
 		// when approxNumPv is 1, most likely a non human curated PV
-		approxNumPv = NonHumanCuratedNumPv
+		approxNumPv = nonHumanCuratedNumPv
 	}
-	rankingInfo := RankingInfo{approxNumPv, tokenString}
+	// Ranking info is only dependent on a stat var (group).
```

Comment:
@chejennifer The refactor is based on this observation. Let me if this makes sense.

Reply:
yes, makes sense